File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/glfw/input/mouse/MouseCursor.java
package org.saar.lwjgl.glfw.input.mouse;

import org.lwjgl.glfw.GLFW;

public enum MouseCursor {

    NORMAL(GLFW.GLFW_CURSOR_NORMAL),
    HIDDEN(GLFW.GLFW_CURSOR_HIDDEN),
    DISABLED(GLFW.GLFW_CURSOR_DISABLED),
    ;

    private static final MouseCursor[] values = values();

    private final int value;

    MouseCursor(int value) {
        this.value = value;
    }

    public int get() {
        return this.value;
    }

    public static MouseCursor valueOf(int value) {
        final int index = value - MouseCursor.values[0].get();
        if (index >= 0 && index <= MouseCursor.values.length) {
            return MouseCursor.values[index];
        }
        throw new IllegalArgumentException("MouseCursor non found: " + value);
    }
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/glfw/input/mouse/MouseCursor.kt
package org.saar.lwjgl.glfw.input.mouse

import org.lwjgl.glfw.GLFW

enum class MouseCursor(private val value: Int) {

    NORMAL(GLFW.GLFW_CURSOR_NORMAL),
    HIDDEN(GLFW.GLFW_CURSOR_HIDDEN),
    DISABLED(GLFW.GLFW_CURSOR_DISABLED),
    ;

    fun get() = this.value

    companion object {
        @JvmStatic
        fun valueOf(value: Int) = when (value) {
            NORMAL.value -> NORMAL
            HIDDEN.value -> HIDDEN
            DISABLED.value -> DISABLED
            else -> throw IllegalArgumentException("MouseCursor non found: $value")
        }
    }
}